src/Database/LSMTree/Internal/Run/Construction.hs

```diff
@@ -43,60 +39,13 @@ module Database.LSMTree.Internal.Run.Construction (
 import           Control.Exception (assert)
 import           Data.Bits (Bits (..))
 import qualified Data.ByteString.Builder as BB
-import           Data.ByteString.Short.Internal (ShortByteString (SBS))
 import           Data.Foldable (Foldable (..))
 import           Data.Maybe (fromMaybe)
 import           Data.Monoid (Dual (..))
-import           Data.Primitive.ByteArray (ByteArray (..), sizeofByteArray)
 import           Data.Word (Word16, Word32, Word64, Word8)
 import           Database.LSMTree.Internal.Entry (Entry (..), onBlobRef,
                      onValue)
-import           Database.LSMTree.Internal.Serialise (SerialisedKey,
-                     serialisedKey, shortByteStringFromTo, sizeofKey16,
-                     sizeofKey64, topBits16)
-
-{-------------------------------------------------------------------------------
-  Types
--------------------------------------------------------------------------------}
-
--- | A string of raw bytes representing a value.
---
--- Can also double as a representation for a chunk of a value.
-data RawValue =
-    UnsafeRawValue !Int !Int ByteArray
-  | EmptyRawValue
-  deriving Show
-
-{-# COMPLETE EmptyRawValue, RawValue #-}
-pattern RawValue :: Int -> Int -> ByteArray -> RawValue
-pattern RawValue start end ba <- UnsafeRawValue start end ba
-  where RawValue start end ba =
-          assert (0 <= start && start <= end && end <= sizeofByteArray ba) $
-          UnsafeRawValue start end ba
-
--- | Size of value in number of bytes.
-sizeofValue :: RawValue -> Int
-sizeofValue EmptyRawValue          = 0
-sizeofValue (RawValue start end _) = end - start
-
--- | Size of value in number of bytes.
-sizeofValue16 :: RawValue -> Word16
-sizeofValue16 = fromIntegral . sizeofValue
-
--- | Size of value in number of bytes.
-sizeofValue32 :: RawValue -> Word32
-sizeofValue32 = fromIntegral . sizeofValue
-
--- | Size of value in number of bytes.
-sizeofValue64 :: RawValue -> Word64
-sizeofValue64 = fromIntegral . sizeofValue
-
--- | TODO: replace this placeholder type by the actual blob reference type.
-data BlobRef = BlobRef {
```

Comment:
I'm not quite sure why we're replacing the `BlobRef` type with `(Word64, Word32)`. Seems like the `BlobRef` was better: more compact, and a distinct type with a clear name.